docs/src/start/embedded.md

```diff
@@ -1 +1,77 @@
 # Using Steel as an embedded scripting engine
+
+Steel can be used as a scripting language within a Rust program. You can achieve
+this by creating a Steel `Engine` object. The `Engine` object allows you to
+execute Scheme code and interact with it from your Rust program. For more
+details about `Engine`, see the [Engine API](../engine/engine.html).
+
+## Steel Engine
+
+The Steel Virtual Machine is provided by the `steel-core` trait.
+
+```toml
+[dependencies]
+steel-core = { git="https://github.com/mattwparas/steel.git", branch = "master" }
+```
+
+The following example runs a few expressions in a Steel `Engine` and asserts
+that the results are as expected.
+
+```rust,noplaypen
+use steel::steel_vm::engine::Engine;
+use steel::SteelVal;
+
+fn main() {
+    let mut steel_engine = Engine::new();
+    let answer = steel_engine.run(
+        (r#"
+      (+ 1 2 3 4)
+      (+ 5 6 7 8)
+    "#),
+    );
+    assert_eq!(answer, vec![SteelVal::IntV(10), SteelVal::IntV(26)])
+}
+```
+
+### Engine::new
+
+Creates a new engine. The `Engine` is used to run Steel Scheme code. Note that
+the `Engine` is not `Send` or `Sync`. This means it is bound to the current
+thread and cannot be shared or sent across other threads.
+
+```rust,noplaypen
+let mut steel_engine = Engine::new();
+```
+
+### Engine::run
+
+Runs a Steel expression and returns the result as a `Vec<SteelVal>`. If any
+error occurs, then `Err(SteelErr)` is returned.
+
+```rust,noplaypen
+let mut steel_engine = Engine::new();
+assert_eq!(steel_engine.run("(+ 1 1)"), Ok(vec![SteelVal::IntV(2)]));
+assert!(steel_engine.run("(+ 1 undefined-identifier)").is_err());
+```
+
+## Embedding The Steel REPL
+
+Repl functionality is provided by the `steel-repl` crate.
+
+```toml
+[dependencies]
+steel-repl = { git="https://github.com/mattwparas/steel.git", branch = "master" }
+```
+
+### repl_base
```

Comment:
Is it ok to stabilize this a bit more? If so, perhaps it can be renamed to `steel_repl::run_repl` or maybe refactored to:

```rust
let repl = steel_repl::Repl::new();
repl.run().unwrap();
```

Reply:
For reference: I think this is the Guile C equivalent: https://www.gnu.org/software/guile/manual/html_node/Initialization.html#index-scm_005fshell

Reply:
Yeah, I agree the name isn't great as is. `run_repl` as a standalone function sounds good, and then maybe a `Repl` struct like you mentioned to configure it a bit more if needed.

Do you want to make that change as part of this? If not - I can make the adjustments in a follow up PR

Reply:
Made it a part of this change.

Reply:
thanks!